pkg/destroy/aws/aws.go

```diff
@@ -342,11 +342,16 @@ func getSharedHostedZone(client *route53.Route53, privateID string, logger logru
 		return "", err
 	}
 
-	if !*response.HostedZone.Config.PrivateZone {
-		return "", errors.Errorf("getShareedHostedZone requires a private ID, but was passed the public %s", privateID)
-	}
 	privateName := *response.HostedZone.Name
 
+	if response.HostedZone.Config != nil && response.HostedZone.Config.PrivateZone != nil {
+		if !*response.HostedZone.Config.PrivateZone {
+			return "", errors.Errorf("getSharedHostedZone requires a private ID, but was passed the public %s", privateID)
+		}
+	} else {
+		logger.Warnf("could not determine whether hosted zone %q is private", privateName)
```

Comment:
nit: this package likes `WithField`.  What do you think about:

```go
logger.WithField("name", privateName).Warn("could not determine whether hosted zone is private")
```

Reply:
"name" does not help much, but I am fine with doing `.WithField("hosted zone", privateName)`

Reply:
> "name" does not help much, but I am fine with doing .WithField("hosted zone", privateName)

I think the thing being named will be clear from the "hosted zone" later in the error message.  We'll probably also have the ARN in there too.  For example, a recent CI run has:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1155/pull-ci-openshift-installer-master-e2e-aws/3328/artifacts/e2e-aws/installer/.openshift_install.log | grep hostedzone/
time="2019-02-01T00:26:46Z" level=debug msg="listing AWS hosted zones (page 0)" arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN
time="2019-02-01T00:26:47Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN public zone=/hostedzone/Z328TAU66YDJH7 record set="A ci-op-zbsntbx0-1d3f3-api.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:47Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="A ci-op-zbsntbx0-1d3f3-api.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:47Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="A ci-op-zbsntbx0-1d3f3-etcd-0.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:47Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="A ci-op-zbsntbx0-1d3f3-etcd-1.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:48Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="A ci-op-zbsntbx0-1d3f3-etcd-2.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:48Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="SRV _etcd-server-ssl._tcp.ci-op-zbsntbx0-1d3f3.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:48Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN public zone=/hostedzone/Z328TAU66YDJH7 record set="A \\052.apps.ci-op-zbsntbx0-1d3f3.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:48Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN record set="A \\052.apps.ci-op-zbsntbx0-1d3f3.origin-ci-int-aws.dev.rhcloud.com."
time="2019-02-01T00:26:48Z" level=info msg=Deleted arn="arn:aws:route53:::hostedzone/Z3BIQ07DC0R5FN" id=Z3BIQ07DC0R5FN
```

Reply:
My understanding of the purpose of fields in the logger is to be able to do filtering based upon the fields. A field of name does not help with filtering as much as a field that distinguishes the type of resource being addressed by the log message.

Additionally, a field of name can be ambiguous. For example, take the log message generated at https://github.com/openshift/installer/blob/4907cba3ded1c014f6583f06b51a2e4cf98e8414/pkg/destroy/aws/aws.go#L1008. There are two resources being addressed in the log message, namely the instance profile and the role. Which resource does "name" reference?